### PR TITLE
Add option to provide test-specific args through host config

### PR DESCRIPTION
**Pull request by harish-24**
Patch adds a way to provide test-specific arguments through config file replacing the existing execution order as the input through config file.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Config file
-------------
```
avocado-misc-tests/memory/ndctl_selftest.py "--keep-tmp on"
avocado-misc-tests/memory/ndctl.py avocado-misc-tests/memory/ndctl.py.data/ndctl.yaml "--mux-filter-only /run/config/version/distro --mux-filter-out /run/config/mode_types -d"
avocado-misc-tests/fs/xfstests.py avocado-misc-tests/fs/xfstests.py.data/nvdimm.yaml "-d"
avocado-misc-tests/fs/xfstests.py avocado-misc-tests/fs/xfstests.py.data/nvdimm_log.yaml "-d"
avocado-misc-tests/io/disk/fiotest.py avocado-misc-tests/io/disk/fiotest.py.data/fio-pmem.yaml
```
Summary
```
# python avocado-setup.py --run-suite host_nvdimm
04:46:28 INFO    : Check for environment
04:46:28 INFO    : Creating temporary mux dir
04:46:31 INFO    : 
04:46:31 INFO    : Running Host Tests Suite nvdimm_ndctl_selftest
04:46:31 INFO    : Running: /usr/bin/avocado run avocado-misc-tests/memory/ndctl_selftest.py --force-job-id 5d44739f77685b8b371c6f0786c55fed4bc1fcc3  --job-results-dir /home/harish/tests/results --keep-tmp on
/usr/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Subscript' object has no attribute 'id'
JOB ID     : 5d44739f77685b8b371c6f0786c55fed4bc1fcc3
JOB LOG    : /home/harish/tests/results/job-2020-03-05T04.46-5d44739/job.log
 (1/1) avocado-misc-tests/memory/ndctl_selftest.py:NdctlTest.test: FAIL: ['FAIL: clear.sh', 'FAIL: daxdev-errors.sh', 'FAIL: rescan-partitions.sh'] (144.67 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 144.71 s
JOB HTML   : /home/harish/tests/results/job-2020-03-05T04.46-5d44739/results.html
04:48:56 INFO    : 
04:48:56 INFO    : 
04:48:56 INFO    : Running Host Tests Suite nvdimm_ndctl_ndctl
04:48:56 INFO    : Running: /usr/bin/avocado run avocado-misc-tests/memory/ndctl.py -m /home/harish/tests/tests/avocado-misc-tests/memory/ndctl.py.data/ndctl.yaml --force-job-id 1711a0b8257a9e4047eec7dd008b434e8f10bbb9  --job-results-dir /home/harish/tests/results --mux-filter-only /run/config/version/distro --mux-filter-out /run/config/mode_types -d
/usr/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 1711a0b8257a9e4047eec7dd008b434e8f10bbb9
JOB LOG    : /home/harish/tests/results/job-2020-03-05T04.48-1711a0b/job.log
 (01/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_bus_ids;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (02/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_dimms;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (03/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_regions;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (04/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (05/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_disable_enable_ns;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (06/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace_modes;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (07/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_namespaces_region;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (08/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_ns_multiple_region;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (09/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_ns_modes_region;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (10/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_nslot_namespace;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (11/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace_reconfigure;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (12/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_check_namespace;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (13/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_check_numa;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (14/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_label_read_write;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (15/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_daxctl_list;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (16/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_sector_write;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (17/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_fsdax_write;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (18/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_map_sync;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (19/19) avocado-misc-tests/memory/ndctl.py:NdctlTest.test_devdax_write;fio_job-distro-1bf6: CANCEL: Test cancelled due to --dry-run (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 19
JOB TIME   : 0.47 s
JOB HTML   : /home/harish/tests/results/job-2020-03-05T04.48-1711a0b/results.html
04:48:57 INFO    : 
04:48:57 INFO    : 
04:48:57 INFO    : Running Host Tests Suite nvdimm_xfstests_nvdimm
04:48:57 INFO    : Running: /usr/bin/avocado run avocado-misc-tests/fs/xfstests.py -m /home/harish/tests/tests/avocado-misc-tests/fs/xfstests.py.data/nvdimm.yaml --force-job-id 665332e26499bddac3746424b039cb175b4811be  --job-results-dir /home/harish/tests/results -d
/usr/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 665332e26499bddac3746424b039cb175b4811be
JOB LOG    : /home/harish/tests/results/job-2020-03-05T04.48-665332e/job.log
 (1/2) avocado-misc-tests/fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-fs_ext4-b619: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (2/2) avocado-misc-tests/fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-exlcude-gen_exclude-1080: CANCEL: Test cancelled due to --dry-run (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 0.08 s
JOB HTML   : /home/harish/tests/results/job-2020-03-05T04.48-665332e/results.html
04:48:58 INFO    : 
04:48:58 INFO    : 
04:48:58 INFO    : Running Host Tests Suite nvdimm_xfstests_nvdimm_log
04:48:58 INFO    : Running: /usr/bin/avocado run avocado-misc-tests/fs/xfstests.py -m /home/harish/tests/tests/avocado-misc-tests/fs/xfstests.py.data/nvdimm_log.yaml --force-job-id efa6d946642b04eef55ea7a421e344f06370d604  --job-results-dir /home/harish/tests/results -d
/usr/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : efa6d946642b04eef55ea7a421e344f06370d604
JOB LOG    : /home/harish/tests/results/job-2020-03-05T04.48-efa6d94/job.log
 (1/2) avocado-misc-tests/fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-log_scratch-log_test-fs_ext4-464c: CANCEL: Test cancelled due to --dry-run (0.00 s)
 (2/2) avocado-misc-tests/fs/xfstests.py:Xfstests.test;disk_scratch-disk_test-log_scratch-log_test-exclude-gen_exclude-8236: CANCEL: Test cancelled due to --dry-run (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 0.08 s
JOB HTML   : /home/harish/tests/results/job-2020-03-05T04.48-efa6d94/results.html
04:48:59 INFO    : 
04:48:59 INFO    : 
04:48:59 INFO    : Running Host Tests Suite nvdimm_fiotest_fio-pmem
04:48:59 INFO    : Running: /usr/bin/avocado run avocado-misc-tests/io/disk/fiotest.py -m /home/harish/tests/tests/avocado-misc-tests/io/disk/fiotest.py.data/fio-pmem.yaml --force-job-id 2e3a1aa129e1e9c805ca6be0b13d335daebfe79d  --job-results-dir /home/harish/tests/results
/usr/lib/python3.6/site-packages/avocado_framework-76.0-py3.6.egg/avocado/plugins/run.py:214: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
JOB ID     : 2e3a1aa129e1e9c805ca6be0b13d335daebfe79d
JOB LOG    : /home/harish/tests/results/job-2020-03-05T04.48-2e3a1aa/job.log
 (1/3) avocado-misc-tests/io/disk/fiotest.py:FioTest.test;dir-disk-xfs-f87a: PASS (65.21 s)
 (2/3) avocado-misc-tests/io/disk/fiotest.py:FioTest.test;dir-disk-ext4-f35a: PASS (26.82 s)
 (3/3) avocado-misc-tests/io/disk/fiotest.py:FioTest.test;dir-disk-filename-a3c0: PASS (77.45 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 169.59 s
JOB HTML   : /home/harish/tests/results/job-2020-03-05T04.48-2e3a1aa/results.html
04:51:50 INFO    : 
04:51:50 INFO    : Summary of test results can be found below:
TestSuite                                                                   TestRun    Summary             
 
host_nvdimm_ndctl_selftest                                                  Run        Successfully executed
/home/harish/tests/results/job-2020-03-05T04.46-5d44739/job.log
 
host_nvdimm_ndctl_ndctl                                                     Not_Run    Unable to find job log file

 
host_nvdimm_xfstests_nvdimm                                                 Not_Run    Unable to find job log file

 
host_nvdimm_xfstests_nvdimm_log                                             Not_Run    Unable to find job log file

 
host_nvdimm_fiotest_fio-pmem                                                Run        Successfully executed
/home/harish/tests/results/job-2020-03-05T04.48-2e3a1aa/job.log
04:51:50 INFO    : Removing temporary mux dir
```

File: avocado-setup.py [hunk 1]
```diff
@@ -487,25 +488,28 @@ def parse_test_config(test_config_file, avocado_bin, enable_kvm):
...
             for list_dic in test_list:
                 if test_dic['name'] == list_dic['name'].split('.')[0]:
                     count += 1
             if count:
                 test_dic['name'] += ".%d" % (count + 1)
             if len(line) > 2:
-                test_dic['args'] = " --execution-order %s " % line[2]
+                test_dic['args'] = " %s" % line[2]
             test_list.append(test_dic)
         if mux_flag == 0:
```
Comment (narasimhan-v) [inline avocado-setup.py]: @harish-24 `if mux_flag == 0 and not test_dic['args']:`
Even mux check can be similar to args in the above

Comment (harish-24) [reply avocado-setup.py]: @narasimhan-v Oh, Thanks for pointing out. Will change and revert

**Comment by harish-24** [comment]
@narasimhan-v Pushed with the changes. Thanks

File: avocado-setup.py [hunk 2]
```diff
@@ -487,27 +489,32 @@ def parse_test_config(test_config_file, avocado_bin, enable_kvm):
                 logger.debug("%s does not exist", test_dic['test'])
                 continue
             if len(line) > 1:
```
Comment (sathnaga) [inline avocado-setup.py]: I guess we need a for loop here to process all items, because current logic misses the line with .yaml and additional args.

Comment (sathnaga) [reply avocado-setup.py]: I see it getting handled in ```if len(line) > 2:
                test_dic['args'] = " --execution-order %s " % line[2]	                arg_flag = 1
                test_dic['args'] = " %s" % line[2]```

but it is better to handle here with additional for loop, what say?

Better to have some comments over the code on what part of config is getting handled for future reference....

Comment (harish-24) [reply avocado-setup.py]: I am not sure if I understand properly. `for` loop to process all items meaning for each of the additional args or for each line?

Comment (sathnaga) [reply avocado-setup.py:491]: I was thinking for additional args, it won't appropriate, ignore first comment, adding comment in the code would help for which part for config file is getting processed.

**Comment by harish-24** [comment]
@sathnaga Pushed, Thanks.

**Pull request merged**